### ISSUE TITLE
Incorrect logits use in accuracy metric?

### ISSUE THREAD

**Issue by rasbt** [PyTorchLightning/metrics#1065]
Hey all, 

in https://torchmetrics.readthedocs.io/en/stable/pages/overview.html?highlight=metriccollection#metriccollection

we pass "logits" into the `accuracy` function. 

![Screen Shot 2022-06-03 at 10 49 03 AM](https://user-images.githubusercontent.com/5618407/171901005-4885afbb-a3b7-4525-bf3c-419f1fce9318.png)

According to the `Accuracy` docs, shouldn't the input be either class labels or probabilities? Like shown in the example here?

```python
>>> import torch
>>> from torchmetrics import Accuracy
>>> target = torch.tensor([0, 1, 2, 3])
>>> preds = torch.tensor([0, 2, 1, 3])
>>> accuracy = Accuracy()
>>> accuracy(preds, target)
tensor(0.5000)
>>>
>>> target = torch.tensor([0, 1, 2])
>>> preds = torch.tensor([[0.1, 0.9, 0], [0.3, 0.1, 0.6], [0.2, 0.5, 0.3]])
>>> accuracy = Accuracy(top_k=2)
>>> accuracy(preds, target)
tensor(0.6667)
```

(https://torchmetrics.readthedocs.io/en/stable/classification/accuracy.html?highlight=accuracy)

**Comment by rasbt** [comment]
Please nevermind, logits are supported :D

**Comment by EklavyaFCB** [comment]
Hi,
Where does it say logits are supported ?

**Comment by rasbt** [comment]
[In the docstring](https://github.com/Lightning-AI/metrics/blob/master/src/torchmetrics/classification/accuracy.py#L52), I found the following

> If preds is a floating point tensor with values outside
      [0,1] range we consider the input to be logits and will auto apply sigmoid per element.

There may be edge cases where you have no negative logits, where this would not work then.

**Comment by EklavyaFCB** [comment]
That's for binary accuracy. For [regular](https://github.com/Lightning-AI/metrics/blob/master/src/torchmetrics/classification/accuracy.py#L347) accuracy, it says one has to use the `top_k` parameter, no ?

**Comment by rasbt** [comment]
Just reading the sentence you referenced:

> For multi-class and multi-dimensional multi-class data with probability or logits predictions, the
    parameter ``top_k`` generalizes this metric to a Top-K accuracy metric: for each sample the
    top-K highest probability or logit score items are considered to find the correct label.

It looks like it supports logits as well. The `top_k` parameter you mentioned is useful for multi-dimensional multi-class classification. Be default  it is `None` / `1`. This is just the regular way you obtain a class label from softmax activations.
